noxfile.py

```diff
@@ -67,48 +67,50 @@ def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> No
-    merged_results = {"benchmarks": []}
-    files = glob.glob("results_*.json")
-    for file in files:
+def valid_results_file(file_name: str) -> bool:
+    """Check if the results file (json) is valid, i.e. existing, non-empty and non-corrupted file."""
+    if not os.path.exists(file_name):
+        return False
+    
+    with open(file_name, "r") as f:
+        content = f.read().strip()  # Remove whitespace
+
+    if not content:  # Empty string
+        return False
+    else:
         try:
-            with open(file, "r") as f:
-                data = json.load(f)
-                merged_results["benchmarks"].extend(data["benchmarks"])
-                # preserve the first file's metadata for a valid pytest-benchmark json file
-                for key in data.keys():
-                    if key != "benchmarks" and key not in merged_results:
-                        merged_results[key] = data[key]
-        except:
-            # Empty file, i.e. no benchmarks
-            continue
-    with open(bencher_json_file_name, "w") as f:
-        json.dump(merged_results, f, indent=4)
+            data = json.loads(content)
+            if not data:  # Empty dictionary or list
+                return False
+            else:
+                return True
+        except json.JSONDecodeError:
+            return False
 
 @nox.session(python=["3.10", "3.11"],
              requires=["benchmark_model-{python}" + f"({subpackage.id})" for subpackage in MODEL_SUBPACKAGE_PATHS])
-def bencher_baseline(session: nox.Session) -> None:
+@nox.parametrize("subpackage", MODEL_SUBPACKAGE_PATHS)
+def bencher_baseline(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
     """Run pytest benchmarks and upload them using Bencher (https://bencher.dev/) (cloud or self-hosted)."""
-    bencher_json_file_name = f"merged_benchmark_results_{session.python}.json"
-    merge_pytest_benchmark_results(bencher_json_file_name)
-
-    session.run(
-        *f"bencher run \
-        --project {os.environ['BENCHER_PROJECT']} \
-        --token {os.environ['BENCHER_API_TOKEN']} \
-        --branch main \
-        --testbed {os.environ['RUNNER']}:{os.environ['SYSTEM_TAG']}:{os.environ['BACKEND']}:{os.environ['GRID']} \
-        --threshold-measure latency \
-        --threshold-test percentage \
-        --threshold-max-sample-size 64 \
-        --threshold-upper-boundary 0.1 \
-        --thresholds-reset \
-        --err \
-        --adapter python_pytest \
-        --file {bencher_json_file_name}".split(),
-        external=True,
-    )
+    bencher_json_file_name = f"results_{session.python}_{subpackage.replace('/', '_')}.json"
+    if valid_results_file(bencher_json_file_name):
+        session.run(
+            *f"bencher run \
+            --project {os.environ['BENCHER_PROJECT']} \
+            --token {os.environ['BENCHER_API_TOKEN']} \
+            --host {os.environ['BENCHER_HOST']} \
```

Comment:
I would expose the `BENCHER_HOST` in the CI pipeline file to make it more transparent where data goes. Also the `BENCHER_PROJECT` could go there. I assume currently their are set in the ci interface?

Reply:
Indeed this is a bit more clear. Although, if we keep both options for now (i.e. cloud & self-hosting), this means that we will have two yml files where we set the above vars. With the current solution (set them in cicd-ext.cscs.ch), we can have just one pipeline file.

Reply:
Ah I understand now how the setup works.